app/helpers.php

```diff
@@ -84,6 +84,28 @@ function format_timestamp_for_display($timestamp)
 }
 
 /**
+ * Generate a unique key for flashing model data to the session.
+ *
+ * @param  \Illuminate\Database\Eloquent\Model  $model
+ * @param  array  $options
+ * @return string
+ */
+function generate_model_flash_session_key($model, $options = [])
+{
+    if ($model instanceof \Gladiator\Models\Competition) {
+        $key = 'contest_' . $model->contest_id . '-' . 'competition_' . $model->id;
```

Comment:
Yeah, that's true. Felt it wouldn't hurt to just be a bit more specific with the keys to avoid conflicts and overriding stuff. :shrug: